### PR TITLE
Update restock.lic to add sigil book restocking and consolidated bank trip

### PR DESCRIPTION
**Pull request by Elysande1218**
Introduces support for filling sigil books from catalog shops using the
two-step ORDER confirmation flow, and extends the purchase_item flow with
the same order_number mechanism for regular items (e.g. society shops).

All default-hometown items and sigil books now share a single count phase,
one bank withdrawal, and one deposit rather than making separate trips per
category. Custom-hometown items continue to use their own trips.

Sigil books support quantity, min_quantity, and container configuration,
nested under the existing restock key.

## Summary by CodeRabbit

* **New Features**
  * Added an alternative two-step order-number purchase flow for restocking.
  * Full sigil-book and sigil-scroll purchase and stow workflow during restocks.

* **Improvements**
  * Restocking split between default-hometown and custom-hometown handling for clearer routing.
  * Bribe padding and cost planning refined; consolidated coin withdrawal for combined purchases.
  * Purchase routing updated so clerk/ask flow takes precedence over order-number.

* **Tests**
  * Expanded coverage for routing, sigil behavior, purchase rounding, and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/elanthia-online/dr-scripts/pull/7392?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: restock.lic [hunk 1]
```diff
@@ -314,6 +342,118 @@ class Restock
...
+          fput('get sigil-scroll from counter')
+        end
+        unless DRCI.put_away_item?("#{purchase['sigil_type']} sigil-scroll", purchase['book'])
+          echo "Failed to add #{purchase['sigil_type']} scroll to #{purchase['book']}"
+          DRCI.stow_hands
+          return
+        end
+      end
+    end
+  end
```
[the remark this answers is not in the document]
Comment (MahtraDR) [reply restock.lic:457]: Fixed in e200ec5c (non-local exit) and 773fc4fd (duplicate spec blocks).

**Non-local exit** -- went with `failed` flag + `break` instead of `throw`/`catch`. Same control flow, avoids the non-local exit lint, and reads more naturally in a method this small. Rubocop is clean.

**Duplicate spec blocks** -- moved the one unique test (`excludes the sigil_books key from regular item parsing`) into the original `#parse_restockable_items` describe block, then deleted both duplicate blocks. 57 examples, 0 failures.

**Comment by MahtraDR** [comment]
@coderabbitai - update review on this pr to include all commits.

**Comment by MahtraDR** [comment]
@coderabbitai where is that re-review that includes all commits up to now?

**Comment by MahtraDR** [comment]
@coderabbitai try again for that review of all commits up to now? Third time's the charm!

**Comment by MahtraDR** [comment]
The original `#purchase_item` describe block (clerk + standard buy) was already removed in 773fc4fd. Only the comprehensive block with all four tests (clerk, order_number, precedence, standard buy) remains. No further changes needed. @coderabbitai
